model/model_training/models/reward_model.py

```diff
@@ -1,68 +1,94 @@
-import os
-from typing import Literal, Optional, Union
+from dataclasses import dataclass
+from typing import Literal, Optional
 
+import torch
 import torch.nn as nn
-from models import get_specific_model
-from transformers import PretrainedConfig, PreTrainedModel
+from transformers import AutoConfig, AutoModelForSequenceClassification
+from transformers.models.gpt_neox.modeling_gpt_neox import GPTNeoXConfig, GPTNeoXModel, GPTNeoXPreTrainedModel
+from transformers.utils import ModelOutput
 
 
-class RewardModelConfig(PretrainedConfig):
-    base_model_name: str
+class GPTNeoXRewardModelConfig(GPTNeoXConfig):
+    model_type = "gpt_neox_reward_model"
+
     pooling: Literal["mean", "last"]
-    model_type = "reward_model"
-    hidden_size: int | None = None  # deepspeed reads this
 
     def __init__(
         self,
-        base_model_name: str = None,
         pooling: Literal["mean", "last"] = "last",
         **kwargs,
     ):
         super().__init__(**kwargs)
-        self.base_model_name = base_model_name
         self.pooling = pooling or "last"
 
-    @classmethod
-    def from_pretrained(
-        cls, pretrained_model_name_or_path: Union[str, os.PathLike], cache_dir: Optional[str] = None, **kwargs
-    ):
-        config, model_kwargs = super().from_pretrained(pretrained_model_name_or_path, **kwargs)
-        if cache_dir:
-            model_kwargs["cache_dir"] = cache_dir
-        return config, model_kwargs
 
+@dataclass
+class GPTNeoXRewardModelOutput(ModelOutput):
+    """
+    Reward model output.
 
-class RewardModel(PreTrainedModel):
-    config_class = RewardModelConfig
+    Args:
+        logits (`torch.FloatTensor` of shape `(batch_size, 1)`):
+            Reward score
+    """
 
-    def __init__(self, config: RewardModelConfig, cache_dir: Optional[str] = None):
-        super().__init__(config=config)
+    logits: torch.FloatTensor = None
 
-        transformer = get_specific_model(
-            config.base_model_name,
-            seq2seqmodel=False,
-            without_head=True,
-            cache_dir=cache_dir,
-        )
 
-        self.transformer = transformer
-        self.out_proj = nn.Linear(transformer.config.hidden_size, 1)
+class GPTNeoXRewardModel(GPTNeoXPreTrainedModel):
+    config_class = GPTNeoXRewardModelConfig
+
+    def __init__(self, config):
+        if type(config) == GPTNeoXConfig:
```

Comment:
I personally prefer `isinstance(config, GPTNeoXConfig)`

Reply:
The direct test was done intentionally, since `GPTNeoXRewardModelConfig` is also derived from `GPTNeoXConfig` ... e.g. the line was added to support the case when a normal non-rm model name is specified. I will add a comment.